### DISCUSSION TITLE
Does Ripes has a console mode?

### DISCUSSION THREAD

**Discussion by alvesfonseca** [mortbopet/Ripes#204] [Q&A]
Is there any switch which allows for pure cli execution?

I mean, something similar to SPIM (for MIPS architecture).

**Comment by mortbopet** [comment]
Nope! but it could definitely be interesting to get something like this up and running. Interactions with the processor model itself is fairly independent of having GUI components enabled, however, some of the telemetry is fairly GUI intertwined.

**Comment by mortbopet** [comment]
Started a branch [here](https://github.com/mortbopet/Ripes/tree/cmdlinemode) to begin this process. My plan is to build the foundation of the support (process input, run the model) and then let the tasks of gathering telemetry, printing output, etc., be open issues for the public to help in developing.

**Comment by mortbopet** [comment]
834f962b1dac8e52e085ee69986ed53ee65bec7b introduced some initial support for this.
Try it out in the [continuous build](https://github.com/mortbopet/Ripes/releases/tag/continuous) e.g.:
> `./Ripes --mode cli --src complexMul.s -t asm --proc "RV32_6S_DUAL" --isaexts M,C --cpi --cycles`

**Reply by alvesfonseca** [reply]
Dear Morten,
Great!
I will try it and will also try to gather some support (devel workforce) from students I advise.
I have teaching interests on RiscV and it would be nice to have a cli-mode on Ripes. 
We intend to use cli-mode in a electronic judge system for students.
The idea is provide programming contests/exercises/assignments focused on RiscV assembly language.
We already do it with SPIM (a cli-mode MIPS simulator).

**Reply by mortbopet** [reply]
Awesome, let me know if you need anything special to support that.
Added support for emitting JSON formatted output, so that would probably be useful for this:
`./Ripes --mode cli --src ${input}.s -t asm --proc "RV32_5S" --isaexts M,C --regs --json --output ${outfile}`
